tests/helpers.test.ts

```diff
@@ -0,0 +1,108 @@
+
+describe("helpers", () => {
+  const settingsFilePath = `${__dirname}/settings.json`;
+
+  beforeEach(() => {
+    closeSync(openSync(settingsFilePath, 'w'));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    if (existsSync(settingsFilePath)) {
+      unlinkSync(settingsFilePath);
+    }
+  });
+
+   test("Verify can load settings.json", () => {
+     const serviceName = "Settings";
+     const expectedSettings = {"foo": "bar"};
+     writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+
+     const actualSettings = helpers.readSettings(serviceName, settingsFilePath);
+
+     expect(actualSettings).toEqual(expectedSettings);
+   });
+
+   test("Verify cannot load settings.json because it does not exist", () => {
+     //eslint-disable-next-line no-console
+     console.log = jest.fn();
+     const serviceName = "Settings";
+     unlinkSync(settingsFilePath);
+
+     const actualSettings = helpers.readSettings(serviceName, settingsFilePath);
+
+     //eslint-disable-next-line no-console
+     expect(console.log).toHaveBeenCalledWith(expect.stringContaining("[Settings] Unable to read the configuration file: ENOENT: no such file or directory"));
+     expect(actualSettings).toBeNull();
+   });
+
+   test("Verify throws if rawConfig empty", () => {
+     const serviceName = "Settings";
+     const expectedSettings = "";
+     writeFileSync(settingsFilePath, expectedSettings);
+
+     expect(() => {helpers.readSettings(serviceName, settingsFilePath)}).toThrow(Error);
+   });
+
+   test("Verify throws with message if rawConfig empty", () => {
+     const serviceName = "Settings";
+     const expectedSettings = "";
+     writeFileSync(settingsFilePath, expectedSettings);
+
+     try {
+       helpers.readSettings(serviceName, settingsFilePath);
+     } catch (error) {
+       expect(error.message).toBe(`[${serviceName}] Unable to load configuration file ${settingsFilePath}.`);
+     }
+   });
+
+  test("Verify throws if json invalid", () => {
+    const serviceName = "Settings";
+    const expectedSettings = {};
+    writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+    const mockAddListener = jest.spyOn(JSONC, 'parse');
+    mockAddListener.mockImplementation((rawConfig, parseErrors) => {
+        parseErrors.push({
+            error: 1,
+            offset: 2,
+            length: 3,
+        });
+        return {};
+    });
+
+    expect(() => {helpers.readSettings(serviceName, settingsFilePath)}).toThrow(Error);
+  });
+
+  test("Verify throws with message if json invalid", () => {
+    const serviceName = "Settings";
+    const expectedSettings = {};
+    writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+    const mockAddListener = jest.spyOn(JSONC, 'parse');
+    mockAddListener.mockImplementation((rawConfig, parseErrors) => {
+        parseErrors = [{
+            error: 1,
+            offset: 2,
+            length: 3,
+        }];
+        return {};
+    });
```

Comment:
I do my best to not mock, but for the life of me I could not get `JSONC` to create parse errors...

Reply:
I'm not surprised 😂 JSONC is super resilient when it comes to parsing errors.

Reply:
Yes that is definitely the case, lol. I personally don't think we need the extra checks here: https://github.com/danecreekphotography/node-deepstackai-trigger/blob/acb23278148d2e18b7f8aaa674e7385578595b28/src/helpers.ts#L36-L42

My approach is typically trusting my gut that I don't need it. If it breaks, hopefully a user will post an issue and details on how to reproduce, then we can turn those reproduction steps into a test and add SUT code that is 100% needed, rather than code that may never get invoked.